service/matching/taskWriter.go

```diff
@@ -201,7 +201,7 @@ func (w *taskWriter) appendTasks(
 		return resp, nil
 
 	case *persistence.ConditionFailedError:
-		w.tlMgr.signalFatalProblem(w.tlMgr.taskQueueID)
+		w.tlMgr.signalFatalProblem(w.tlMgr)
```

Comment:
BTW, if we were just call w.tlMgr.Unload() here we won't have to deal with this bug.

Reply:
Only if Unload dedups. Which it doesn't. Stop() here would work because it dedups for any single taskQueueManager instance.

Reply:
tlMgr.Stop maybe? that would work because it dedups on a per-taskQueueManager instance basis.

Reply:
I meat tlm.Stop(), but yes we still need to remove it from the collection of matching engine, which is keyed on taskQueueID, and dedupe would still needed. So, maybe it does not matter.